packages/data-schema/src/SchemaProcessor.ts

```diff
@@ -974,15 +1008,19 @@ function processFields(
   fields: Record<string, any>,
   impliedFields: Record<string, any>,
   fieldLevelAuthRules: Record<string, string | null>,
+  getRefType: ReturnType<typeof getRefTypeForSchema>,
+
   identifier?: readonly string[],
   partitionKey?: string,
   secondaryIndexes: TransformedSecondaryIndexes = {},
   databaseEngine: DatasourceEngine = 'dynamodb',
+  generateInputType: boolean = false,
```

Comment:
Could the generation of input types be handled by another function instead of overloading this function?